bundle/artifacts/artifacts_test.go

```diff
@@ -84,26 +100,39 @@ func TestArtifactUploadForWorkspace(t *testing.T) {
 		},
 	}
 
-	artifact := b.Config.Artifacts["whl"]
-	mockFiler := mockfiler.NewMockFiler(t)
-	mockFiler.EXPECT().Write(
+	whlMockFiler := mockfiler.NewMockFiler(t)
+	whlMockFiler.EXPECT().Write(
 		mock.Anything,
 		filepath.Join("source.whl"),
 		mock.AnythingOfType("*bytes.Reader"),
 		filer.OverwriteIfExists,
 		filer.CreateParentDirectories,
 	).Return(nil)
 
-	err := uploadArtifact(context.Background(), b, artifact, "/foo/bar/artifacts", mockFiler)
-	require.NoError(t, err)
+	whlError := uploadArtifact(context.Background(), b, b.Config.Artifacts["whl"], "/foo/bar/artifacts", whlMockFiler)
+	require.NoError(t, whlError)
+
+	requirementsMockFiler := mockfiler.NewMockFiler(t)
+	requirementsMockFiler.EXPECT().Write(
+		mock.Anything,
+		filepath.Join("requirements.txt"),
+		mock.AnythingOfType("*bytes.Reader"),
+		filer.OverwriteIfExists,
+		filer.CreateParentDirectories,
+	).Return(nil)
+
+	requirementsError := uploadArtifact(context.Background(), b, b.Config.Artifacts["requirements"], "/foo/bar/artifacts", requirementsMockFiler)
```

Comment:
I think this test is passing only because you added the following section in the test but that's not something users will configure

```
"requirements": {
					Type: config.ArtifactPythonRequirementsFile,
					Files: []config.ArtifactFile{
						{Source: requirementsLocalPath},
					},
				},
```